### PR TITLE
Fixed bugs

### PR DESCRIPTION
**Pull request by SannihithReddy610**

### CODE REVIEW

File: functiondefextractor/condition_checker.py [hunk 1]
```diff
@@ -67,11 +68,15 @@ def get_pivot_table_result(data, test_assert, splitter, file_path):
     data_table = data_table.to_frame()
     data_table = data_table.reset_index()
     data_table = data_table.rename({'index': 'Different %s patterns ' % test_assert}, axis='columns')
+    if data_table.iat[0, 0] == '':  # pragma: no mutate
+        data_table = data_table.drop([data_table.index[0]])  # pragma: no mutate
```
Comment (BrijeshKrishnan) [inline functiondefextractor/condition_checker.py]: What significance with this check here ?

File: functiondefextractor/core_extractor.py [hunk 2]
```diff
@@ -23,15 +23,9 @@ def get_file_names(dir_path):
...
-    for entry in listoffile:
-        fullpath = os.path.join(dir_path, entry)
-        if os.path.isdir(fullpath):
-            allfiles = allfiles + get_file_names(fullpath)
-        else:
-            allfiles.append(fullpath)
-    return allfiles
+    path_list = [os.path.join(dirpath, filename) for dirpath, _, filenames in
+                 os.walk(dir_path) for filename in filenames]
+    return path_list
```
Comment (BrijeshKrishnan) [inline functiondefextractor/core_extractor.py]: Thanks, Looks cleaner now!

File: functiondefextractor/core_extractor.py [hunk 3]
```diff
@@ -230,8 +245,8 @@ def check_py_annot(file_name, annot):
     line_data = list(
         [line.rstrip() for line in open(file_name, encoding='utf-8', errors='ignore')])  # pragma: no mutate
     val = 1  # pragma: no mutate
-    if annot.upper() == "TEST_":  # Making use of annotation search function for function start with feature too
-        annot = "def test_"  # pragma: no mutate
+    if annot[0] != "@":  # Making use of annotation search function for function start with feature too
+        annot = "def " + annot.lower()  # pragma: no mutate
```
Comment (BrijeshKrishnan) [inline functiondefextractor/core_extractor.py]: Thanks for making in generic

File: test/test_core_extractor.py [hunk 4]
```diff
@@ -90,6 +90,7 @@ def test_process_python_test_extract(self):
         """Function to test the complete end to end process of function definition extractor (True True)"""
         dataframe = extractor((os.path.join(self.file_path, "test_resource", "test_repo")), functionstartwith="test_",
                               report_folder=None)
+        dataframe.to_excel("teee.xlsx")
```
Comment (BrijeshKrishnan) [inline test/test_core_extractor.py]: Was it by mistake? may be can be removed!

File: test/test_core_extractor.py [hunk 5]
```diff
@@ -173,7 +155,7 @@ def test_cmd_inputs(self):
 
     def test_extractor_cmd(self):
         """Function to test command line working"""
-        cmd = 'python -m functiondefextractor.extractor_cmd --p "%s"' \
+        cmd = 'python -m functiondefextractor --p "%s"' \
```
Comment (BrijeshKrishnan) [inline test/test_core_extractor.py]: Please make sure not to over wright the files. This change was incorporated in the previous merge !

**Review by BrijeshKrishnan — CHANGES_REQUESTED** [review]

File: functiondefextractor/condition_checker.py (hunk 1 above)
Comment (SannihithReddy610) [reply functiondefextractor/condition_checker.py]: Thanks for the suggestion. It's is nor required. Removed

File: test/test_core_extractor.py (hunk 4 above)
Comment (SannihithReddy610) [reply test/test_core_extractor.py]: Yes it's by mistake. Corrected now

File: test/test_core_extractor.py (hunk 5 above)
Comment (SannihithReddy610) [reply test/test_core_extractor.py]: Corrected

**Review by BrijeshKrishnan — APPROVED** [review]

**Review by BrijeshKrishnan — APPROVED** [review]

**Pull request merged**